bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,309 @@
+        codewars.com API to use.
+        """
+        async with self.bot.http_session.get(search_link, params=params) as response:
+            if response.status != 200:
+                logger.error(
+                    f"Unexpected status code {response.status} from codewars.com"
+                )
+            soup = BeautifulSoup(await response.text(), features="lxml")
+            first_kata_div = soup.find_all("div", class_="item-title px-0")
+
+            if not first_kata_div:
+                raise commands.BadArgument("No katas could be found with the filters provided.")
+            elif len(first_kata_div) >= 3:
+                first_kata_div = choice(first_kata_div[:3])
+            elif 'q=' not in search_link:
+                first_kata_div = choice(first_kata_div)
+            else:
+                first_kata_div = first_kata_div[0]
+
+            # there are numerous divs before arriving at the id of the kata, which can be used for the link.
+            first_kata_id = first_kata_div.a["href"].split("/")[-1]
+            return first_kata_id
+
+    async def kata_information(self, kata_id: str) -> dict:
+        """
+        Returns the information about the Kata.
+
+        Uses the codewars.com API to get information about the kata using the kata's ID.
+        """
+        async with self.bot.http_session.get(API_ROOT.format(kata_id=kata_id)) as response:
+            if response.status != 200:
+                logger.error(
+                    f"Unexpected status code {response.status} from codewars.com/api/v1"
+                )
+            kata_information = await response.json()
+            return kata_information
+
+    @staticmethod
+    def main_embed(kata_information: dict) -> Embed:
+        """
+        Creates the main embed which displays the description of the kata and the difficulty, along with the name.
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
+        """
+        # maps difficulty to color for embed
+        mapping_of_kyu = {
+            8: (221, 219, 218), 7: (221, 219, 218), 6: (236, 182, 19), 5: (236, 182, 19),
+            4: (60, 126, 187), 3: (60, 126, 187), 2: (134, 108, 199), 1: (134, 108, 199)
+        }
+
+        kata_description = kata_information["description"]
+
+        # ensuring it isn't over the length 1024
+        if len(kata_description) > 1024:
+            kata_description = "\n".join(kata_description[:1000].split("\n")[:-1])
+            kata_id = kata_information["id"]
+            kata_description += f"\n[Read more]({f'https://codewars.com/kata/{kata_id}'})"
+
+        # creating the main kata embed
+        kata_embed = Embed(
+            title=kata_information['name'],
+            description=kata_description,
+            color=Color.from_rgb(
+                *mapping_of_kyu[int(kata_information['rank']['name'].replace(' kyu', ''))]
+            ),
+        )
```

Comment:
Can we add a URL to this embed so that the title is clickable as well?

Even though you do get the link with a button.

Reply:
Sure!